pkg/client/cache/listers.go

```diff
@@ -69,10 +69,17 @@ func (s *StoreToNodeLister) List() (machines api.NodeList, err error) {
 // rather than a method of StoreToNodeLister.
 // GetNodeInfo returns cached data for the minion 'id'.
 func (s *StoreToNodeLister) GetNodeInfo(id string) (*api.Node, error) {
-	if minion, ok := s.Get(id); ok {
-		return minion.(*api.Node), nil
+	minion, exists, err := s.Get(&api.Node{ObjectMeta: api.ObjectMeta{Name: id}})
```

Comment:
Nodes don't have namespaces, and won't.  So until that happens, node name is sufficient.

Maybe not understanding.

----- Original Message -----
> > @@ -69,10 +69,17 @@ func (s *StoreToNodeLister) List() (machines
> > api.NodeList, err error) {
> >  // rather than a method of StoreToNodeLister.
> >  // GetNodeInfo returns cached data for the minion 'id'.
> >  func (s *StoreToNodeLister) GetNodeInfo(id string) (*api.Node, error) {
> > -	if minion, ok := s.Get(id); ok {
> > -		return minion.(*api.Node), nil
> > +	minion, exists, err := s.Get(&api.Node{ObjectMeta: api.ObjectMeta{Name:
> > id}})
> 
> @smarterclayton So, this replaces all the interface changes, *only if* Nodes
> have an empty namespace. If they are stored with some uniform non-blank
> namespace, either:
> 
> 1. That namespace would have to be hard-coded here or,
> 2. The Node would need to propagate down to here via the interfaces.
> 
> Thoughts?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/3810/files#r23789908